### DISCUSSION TITLE
GetSocket has no way to set Headers

### DISCUSSION THREAD

**Discussion by khuntia** [jonataslaw/getx#1219] [Ideas]
Under the basewebsocket, for getsocket, we only have provision to pass the url of the socket,
not additional headers. This makes it restrictive as I cant pass authentication headers or other oauth 
headers when connecting to the backend. Please add provision for passing headers as well

//this is how it is right now
socket = WebSocket(url); 

//Rather we do WebSocket(url, Map headers)

**Comment by jonataslaw** [comment]
GetConnect is multiplatform, it is not possible to set headers on the web, I do not know if we should provide this only for the desktop, but we can do this, so I am moving this to discussions

**Reply by subzero911** [reply]
I looked at this https://pub.dev/documentation/web_socket_channel/latest/web_socket_channel/WebSocketChannel-class.html
I found they have static signKey method you can use to sign up a security header.
It could be like:
```dart
final socket = GetSocket(url);
if (!platfrom.isWeb) {
   socket.signKey('aabbcc123');
}
```

**Comment by laeo** [comment]
I'd read [RFC6455](https://www.rfc-editor.org/rfc/rfc6455), it says:

> 12.  The request MAY include any other header fields, for example,
        cookies [[RFC6265](https://www.rfc-editor.org/rfc/rfc6265)] and/or authentication-related header fields
        such as the |Authorization| header field [[RFC2616](https://www.rfc-editor.org/rfc/rfc2616)], which are
        processed according to documents that define them.

So i don't know why there is a compatibility issue.
